common/src/androidTest/java/com/microsoft/identity/common/MicrosoftStsAuthorizationRequestTests.java

```diff
@@ -0,0 +1,180 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common;
+
+import android.support.annotation.NonNull;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.internal.providers.microsoft.microsoftsts.MicrosoftStsAuthorizationRequest;
+import com.microsoft.identity.common.internal.providers.microsoft.microsoftsts.MicrosoftStsPromptBehavior;
+import com.microsoft.identity.common.internal.providers.oauth2.PkceChallenge;
+import com.microsoft.identity.common.internal.util.StringUtil;
+
+import org.junit.Test;
+
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.junit.Assert.assertTrue;
+
+public class MicrosoftStsAuthorizationRequestTests {
+    private static final String DEFAULT_TEST_RESPONSETYPE = "code";
+    private static final String DEFAULT_TEST_CLIENT_ID = "some-client-id";
+    private static final String DEFAULT_TEST_REDIRECT_URI = "some://redirect.uri";
+    private static final String DEFAULT_TEST_STATE = "someState";
+    private static final String DEFAULT_TEST_LOGIN_HINT = "someLoginHint";
+    private static final String[] DEFAULT_TEST_SCOPE_ARRAY = {"scope1", "scope2"};
+    private static final String[] DEFAULT_TEST_EXTRA_SCOPE_ARRAY = {"scope3", "scope4"};
+    private static final Set<String> DEFAULT_TEST_SCOPE = new HashSet<>(Arrays.asList(DEFAULT_TEST_SCOPE_ARRAY));
+    private static final String DEFAULT_TEST_AUTHORIZATION_ENDPOINT = "https://login.microsoftonline.com/common/oauth2/authorize";
+    private static final UUID DEFAULT_TEST_CORRELATION_ID = UUID.randomUUID();
+    private static final String DEFAULT_TEST_EXTRA_QP = "extra=1&haschrome=1";
+    private static final String DEFAULT_TEST_VERSION = "0.1";
+    private static final MicrosoftStsPromptBehavior DEFAULT_TEST_PROMPT = MicrosoftStsPromptBehavior.CONSENT;
+    private static final String DEFAULT_TEST_UID = "1";
+    private static final String DEFAULT_TEST_UTID = "1234-5678-90abcdefg";
+    private static final String DEFAULT_TEST_DISPLAYABLEID = "user@contoso.com";
+    private static final String DEFAULT_TEST_SLICE_PARAMETER = "slice=myslice";
+    private static final Set<String> DEFAULT_TEST_EXTRA_SCOPE = new HashSet<>(Arrays.asList(DEFAULT_TEST_EXTRA_SCOPE_ARRAY));
+    private static final String DEFAULT_TEST_AUTHORITY_STRING = "https://login.microsoftonline.com/common";
+
+    static URL getValidRequestUrl() throws MalformedURLException {
+        return new URL(DEFAULT_TEST_AUTHORITY_STRING);
+    }
+
+    static MicrosoftStsAuthorizationRequest createAuthenticationRequest(final String responseType,

```

Comment:
Why is a method called createAuthenticationRequest returning an Authorization Request?  Any reason to prefer a static method over an actual constructor?  Should we consider having the MicrosoftSTSOAuth2Strategy provide methods for creating the authorization request.